File: csharp/src/DarkStrata.CredentialCheck/Exceptions/NetworkException.cs
using System;

namespace DarkStrata.CredentialCheck;

/// <summary>
/// Exception thrown when a network error occurs.
/// </summary>
public sealed class NetworkException : DarkStrataException
{
    /// <summary>
    /// Creates a new NetworkException.
    /// </summary>
    /// <param name="message">The error message.</param>
    /// <param name="innerException">The inner exception (if any).</param>
    public NetworkException(string message, Exception? innerException = null)
        : base(message, ErrorCode.NetworkError, isRetryable: true, innerException: innerException)
    {
    }
}

File: sdks/java/src/main/java/io/darkstrata/credentialcheck/exception/NetworkException.java
package io.darkstrata.credentialcheck.exception;

/**
 * Exception thrown when a network error occurs.
 */
public class NetworkException extends DarkStrataException {

    public NetworkException(String message) {
        this(message, null);
    }

    public NetworkException(String message, Throwable cause) {
        super(message, ErrorCode.NETWORK_ERROR, null, true, cause);
    }
}
